debezium-core/src/main/java/io/debezium/data/VariableScaleDecimal.java

```diff
@@ -70,11 +71,25 @@ public static Schema optionalSchema() {
      * @return the encoded value
      */
     public static Struct fromLogical(Schema schema, SpecialValueDecimal value) {
+        return fromLogical(schema, value.getDecimalValue().orElse(null));
+    }
+
+    /**
+     * Converts a value from its logical format to its encoded format - a struct containing
+     * the scale of the number and a binary representation of the number.
+     *
+     * @param schema of the encoded value
+     * @param value the value or the decimal
+     *
+     * @return the encoded value
+     */
+    public static Struct fromLogical(Schema schema, BigDecimal decimalValue) {
+        Objects.requireNonNull(decimalValue, "decimalValue may not be null");
+
         Struct result = new Struct(schema);
-        final BigDecimal decimalValue = value.getDecimalValue().orElse(null);
-        assert decimalValue != null : "Unable to encode special value";
```

Comment:
Note I've replaced the assert call by `requireNonNull`. I generally find asserts not overly useful. If we really want to  be sure that the value isn't null, let's enforce it without relying on any runtime parameters like _-ea_.